logger.go

```diff
@@ -21,89 +15,33 @@ type Logger struct {
 	Stderr io.Writer
 }
 
-type PrefixedWriter struct {
-	io.Writer
-	Prefix    string
-	Separator string
+type eventLogger struct {
+	executor *Executor
+	task     *Task
+	stream   TaskLogEventStream
 }
 
-var colors = []color.Attribute{
-	color.FgGreen,
-	color.FgYellow,
-	color.FgBlue,
-	color.FgMagenta,
-	color.FgCyan,
+func (l *eventLogger) Write(p []byte) (int, error) {
+	l.executor.publishEvent(&TaskLogEvent{
+		simpleEvent: l.executor.taskExecution(l.task).simpleEvent(),
+		Message:     string(p),
+		Stream:      l.stream,
+	})
+	return len(p), nil
 }
 
-var defaultPadding = 16
-
-func SetDefaultPadding(padding int) {
-	defaultPadding = padding
-}
-
-func (w *PrefixedWriter) Write(p []byte) (n int, err error) {
-	separator := w.Separator
-	if separator == "" {
-		separator = "|"
+func ProvideEventLogger(e *Executor, t *Task) *Logger {
```

Comment:
godoc comment? can we put this on as a method for the executor?

Reply:
Made it private instead, moved to method on executor